rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -271,14 +271,16 @@ def run_all_post_decoding_rules(
             transaction: EvmTransaction,
             decoded_events: list[HistoryBaseEntry],
             all_logs: list[EvmTxReceiptLog],
-    ) -> None:
+    ) -> list[HistoryBaseEntry]:
         """
         Runs all post-decoding rules from self.rules.post_decoding_rules.
         The post-decoding rules list consists of tuples (priority, rule) and must be sorted by
         priority in ascending order. The higher the priority number the later the rule is run.
         """
         for (_, rule) in self.rules.post_decoding_rules:
-            rule(transaction=transaction, decoded_events=decoded_events, all_logs=all_logs)  # noqa: E501
+            decoded_events = rule(transaction=transaction, decoded_events=decoded_events, all_logs=all_logs)  # noqa: E501
```

Comment:
not related to this PR but the priority is not taken into account right now?

Reply:
It is https://github.com/rotki/rotki/blob/develop/rotkehlchen/chain/evm/decoding/decoder.py#L153